lib/transport/shell.js

```diff
@@ -109,7 +109,7 @@ Shell.prototype.transfer = function(files, remoteDir, options) {
                               (remote.username ? remote.username + '@' : ''),
                               remote.host, remoteDir);
 
-    var command = util.format('rsync --files-from %s %s --rsh="ssh -p%s%s" ./ %s',
+    var command = util.format('cat %s | xargs -n 50 -J % rsync %s --rsh="ssh -p%s%s" % %s',
```

Comment:
A couple of questions:
- Why are you limiting this to 50 files?
- What are the percentage signs doing?
- Is this working with spaces and special characters in file names?
- Is this still working properly with relative paths?
- How does this perform when uploading hundreds of files?